mdbenchmark/generate.py

```diff
@@ -103,10 +115,40 @@ def generate(name, gpu, module, host, min_nodes, max_nodes, time, list_hosts):
             'If you use the {} option make sure you use the GPU compatible NAMD module!',
             '--gpu')
 
+    # Grab list of all available and supported modules
+    available_modules = get_available_modules()
```

Comment:
why not filter the list modules here?

Reply:
Do you mean we should duplicate the `for m in module:` code? Otherwise we cannot do the validation of module names, as we need to iterate through the list.

Reply:
```python
modules = [m for m in modules if m in available_modules]
```

Now all modules are actually available. If you want to warn the user for missing ones you can so

```python
requested_modules = set(modules)
modules = [...]
if requested_modules.difference(modules):
    console.warn(" {} are not available modules".format(requested_modules.difference(modules))

```

This if-clause should only be activated if the difference is not empty. If it is empty all modules are available.